RobotCrawler/src/main/java/parser/ebookscom/EBooksDataCollector.java

```diff
@@ -0,0 +1,45 @@
+package parser.ebookscom;
+
+import org.jsoup.nodes.Element;
+import parser.BookDataCollector;
+
+/**
+ * To collect the data that the {@link EBooksParser}
+ * parsed earlier.
+ */
+public class EBooksDataCollector implements BookDataCollector<Element> {
+
+    public String urlFrom(Element element){
+        return element.select(".book-title a").attr("href");
+    }
+
+    @Override
+    public String libraryFrom(Element element) {
+        return "http://www.ebooks.com/";
```

Comment:
"http://www.ebooks.com/" should be in some final field. I see that this method require some not used parameter. Maybe this method in  abstract  class EBooksDataCollector should be changed.